react/week4/my-app/src/Context.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+
+const Context = React.createContext();
+
+function ContextProvider({ children }) {
+  const [userInput, setUserInput] = useState("");
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  useEffect(() => {
+    if (!userInput || error) {
```

Comment:
Doesn't the logic here mean that if you ever get an error then you will never make a new request to the GitHub API?
Could you explain the idea behind that?

Reply:
no It was meant to be if we don't have  user input or error give us set loading false :/